File: Builder/Builder.Main/Order.cs
using System;

namespace Builder.Main
{
    

    public class Order
    {
        internal class OrderBuilder
        {
            private Guid _customerId;
            private decimal _amount;
            private DateTime _createdAt;

            internal OrderBuilder WithCustomerId(string customerId)
            {
                _customerId = Guid.Parse(customerId);
                return this;
            }

            internal OrderBuilder WithAmount(string amount)
            {
                _amount = decimal.Parse(amount);
                return this;
            }
            internal OrderBuilder WithCreatedAt(string createdAt)
            {
                _createdAt = DateTime.Parse(createdAt);
                return this;
            }
            internal Order Build()
            {
                if (_createdAt == null)
                {
                    _createdAt = DateTime.Now;
                }
                return new Order(_customerId, _amount, _createdAt);
            }
        }
        public Order(Guid customerId, decimal amount, DateTime createdAt)
        {
            CustomerId = customerId;
            Amount = amount;
            CreatedAt = createdAt;
        }

        internal static OrderBuilder CreateBuilder()
        {
            return new OrderBuilder();
        }

        public Guid CustomerId { get; private set; }
        public decimal Amount { get; private set; }
        public DateTime CreatedAt { get; private set; }
    }
}

File: Builder/Builder.Js/Order.js
let Order = function(customerId, amount, createdAt) {
    this.customerId = customerId;
    this.amount = amount;
    this.createdAt = createdAt;
}

let OrderBuilder = function () {
    let customerId;
    let amount;
    let createdAt;
    return {
        withCustomerId: function (customerId) {
            this.customerId = customerId;
            return this;
        },
        withAmount: function (amount) {
            this.amount = amount;
            return this;
        },
        withCreatedAt: function (createdAt) {
            this.createdAt = createdAt;
            return this;
        },
        build: function () {
            return new Order(this.customerId, this.amount, this.createdAt);
        }
    }
}
console.log(new Order(12345678, 100.0, new Date(2019, 5, 12)));

let task = new OrderBuilder()
    .withCustomerId('12345678')
    .withAmount('100.0') 
    .withCreatedAt(new Date(2019, 5, 12))
    .build();
console.log(task);
